src/bootstrap.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+use Silex\Application;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+
+require_once __DIR__.'/../vendor/autoload.php';
+
+$app = new Application();
+
+$app->get('ping', function () {
+    return new Response(
+        'pong',
+        Response::HTTP_OK,
+        [
+            'Cache-Control' => 'must-revalidate, no-cache, no-store, private',
+            'Content-Type' => 'text/plain; charset=UTF-8',
+        ]
+    );
+});
+
+$app->get('/oauth2/authorize', function (Request $request) {
+    $redirectUri = $request->get('redirect_uri');
+    $state = $request->get('state');
+    $code = 'code_'.$state;
+    $location = $redirectUri.'?'.http_build_query([
+        'code' => $code,
+        'state' => $state,
+    ]);
+
+    return new Response(
+        'pong',
+        Response::HTTP_FOUND,
+        [
+            'Location' => $location,
+        ]
+    );
+});
+
+$app->post('/oauth2/token', function (Request $request) {
+    $code = $request->get('code');
```

Comment:
Does ORCID work with `GET` or just `POST`?

Reply:
It's not idempotent and the RFC seems to mandate 'POST (https://tools.ietf.org/html/rfc6749#section-6) so I'd stick to that